feat: Implement to change member grade

jinhokim98 opened the pull request:
# 회원 정보 중 등급 수정 기능

## 주요 변경 내용
- 회원 정보 중 등급을 바꾸는 코드를 구현했습니다.
- 확인메세지 내용을 모았습니다.
- useEffect가 여러 번 호출되는 현상을 수정했습니다.
- 상세 페이지에서 등급을 변경하면 회원정보 탭에서도 변경된 등급이 반영됩니다.

## 참고 사항
- 회원 정보 중 체크박스가 필요없다는 생각이 들었습니다. 이 부분에 대한 논의가 필요할 것 같습니다.

## 남은 작업 내용
- 회원 상세정보에서 리덕스에 저장된 값을 불러오는 코드를 구현할 예정입니다.
- 페이지가 재로딩되면 state가 날라가는 모습을 보입니다. 이것을 해결할 예정입니다.

## 참고용 시연 사진
![image](https://user-images.githubusercontent.com/81083461/222700536-6a89ba51-548c-49b5-989d-a3db1b29c1eb.png)
![image](https://user-images.githubusercontent.com/81083461/222701173-78e30dd9-38c0-4eb0-975b-032f555643bc.png)
![image](https://user-images.githubusercontent.com/81083461/222881660-6930bb1b-d503-44c6-9b88-a1820a6c690d.png)
![image](https://user-images.githubusercontent.com/81083461/222881676-9eb4b20a-0e4f-427d-a930-2e6ebaf4e3e7.png)
![image](https://user-images.githubusercontent.com/81083461/222881702-63fb1a4d-3206-4ef4-aa76-de9aa51a9b18.png)

jinhokim98 committed:
feat: Implement to change member grade

회원 정보 중 등급을 바꾸는 코드를 구현했습니다.
확인메세지 내용을 모았습니다.
useEffect가 여러 번 호출되는 현상을 수정했습니다.

Guardian-demoN reviewed (approved):
기본 문법 준수 및 코드룰 준수 여부 기준으로 코드 확인했습니다.
기능 개발 디테일에 대해서는 차차 확인해볼 생각이며, 이의 있는 경우 Issue를 통해 전달 드리도록 하겠습니다.

jinhokim98 committed:
feat: Show member detail in redux state

회원 상세정보 페이지에서 리덕스에 저장된 상태값을 가져와 보여줍니다.
하지만 페이지를 재로딩하면 state가 날라가는 현상이 보입니다.
이를 해결하기위해 redux-persist를 사용해보려했으나 오류가 있어 우선은 추후에 적용할 예정입니다.
회원정보를 수정하면 수정완료 알림창을 띄웁니다.

jinhokim98 committed:
feat: Implement to change member grade in member detail page

회원 상세페이지에서 등급을 수정한 후 저장을 누르면 등급이 변경됩니다.
저장을 누르면 관리자페이지로 이동하며 회원정보 탭을 누르면 변경한 값이 보여집니다.

jinhokim98 committed:
refactor: Modify reducer name

ChangeGradeReducer에서 ChangeMemberInfoReducer로 변경했습니다.
회원 삭제 기능을 넣기 위해서 이름을 바꾸는 것이 좋다고 생각했습니다.

jinhokim98 committed:
feat: Implememt to get out user

회원 강퇴 상태를 구현했습니다.
